### ISSUE TITLE
Enter toggles fullscreen

### ISSUE THREAD

**Issue by kartun83** [FreeRDP/FreeRDP#5342]
**Describe the bug**
Enter exits FullScreen mode. So, to be clear, when freerdp is in fullscreen mode and I press only Enter, it exits fullscreen, but when in window mode I press enter  it doesn't switch to fullscreen mode.

**To Reproduce**
1. Connect
2. Switch to fullscreen mode with ctrl+alt+enter
3. Type anything and press Enter

**Expected behavior**
Only Ctrl+Alt+Enter should toggle fullscreen mode

**Application details**
* Version of FreeRDP : FreeRDP version 2.0.0-dev5 (df280a7ff)
* Command line used : 
* output of `/buildconfig`
/opt/freerdp-nightly/bin/xfreerdp /v:/build-config
[16:17:55:978] [28135:28136] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[16:17:55:979] [28135:28136] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[16:17:55:979] [28135:28136] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr

* OS version connecting to
Windows 10

**Desktop (please complete the following information):**
 - OS: Linux Mint 19.1

**Command Line**
#!/bin/bash
/opt/freerdp-nightly/bin/xfreerdp /drive:home,/home/XXX/Share /u:XXXX /v:none.com /w:1920 /h:1080 +auto-reconnect /auto-reconnect-max-retries:5 /p:XXX +clipboard /log-level:trace

**Comment by akallabeth** [comment]
Not seen such behaviour so far.
You're using our nightly builds I assume?
Which desktop environment dou you use, it is most likely related to that. (Key bindings, special keys?)

**Comment by kartun83** [comment]
> Not seen such behaviour so far.
> You're using our nightly builds I assume?
> Which desktop environment dou you use, it is most likely related to that. (Key bindings, special keys?)

It's Cinnamon 4.0.10, with default key bindings, no modifications against stock.
Any suggestions how to debug this issue ?

**Comment by kartun83** [comment]
Additionally, here is another issue with fullscreen:
if I use **/f /floatbar** , than ctrl+alt+enter toggles visibility of float bar, in other words without /flotabar I don't see any option to close rdp, and nothing helpful in help =(

**Comment by Ionic** [comment]
It sounds like somehow Ctrl-Alt got stuck in your session. I'm seeing this quite often when switching workspaces, but that wouldn't work in fullscreen mode.

Try pressing Ctrl-Alt after switching to fullscreen mode. Does pressing Enter still take you out of fullscreen mode afterwards?

**Comment by acruxcz** [comment]
Exactly same issue at FreeRDP version 2.0.0-dev5 (5d804852b)
Linux Mint 19.1

**Comment by parshin** [comment]
Try to use /smart-sizing

**Comment by CasparLi** [comment]
> Try to use /smart-sizing

It works!

**Comment by ifconfig-me** [comment]
Thanks, the **/smart-sizing** works when hit ctrl+alt+ENTER.

**Comment by dmsuhanov** [comment]
> **Describe the bug**
> Enter exits FullScreen mode. So, to be clear, when freerdp is in fullscreen mode and I press only Enter, it exits fullscreen, but when in window mode I press enter it doesn't switch to fullscreen mode.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Connect
> 2. Switch to fullscreen mode with ctrl+alt+enter
> 3. Type anything and press Enter
> 
> **Expected behavior**
> Only Ctrl+Alt+Enter should toggle fullscreen mode
> 
> **Application details**
> 
> * Version of FreeRDP : FreeRDP version 2.0.0-dev5 ([df280a7](https://github.com/FreeRDP/FreeRDP/commit/df280a7ffd7dada307942f5f15318b25fe637336))
> * Command line used :
> * output of `/buildconfig`
>   /opt/freerdp-nightly/bin/xfreerdp /v:/build-config
>   [16:17:55:978] [28135:28136] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
>   [16:17:55:979] [28135:28136] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
>   [16:17:55:979] [28135:28136] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
> * OS version connecting to
>   Windows 10
> 
> **Desktop (please complete the following information):**
> 
> * OS: Linux Mint 19.1
> 
> **Command Line**
> #!/bin/bash
> /opt/freerdp-nightly/bin/xfreerdp /drive:home,/home/XXX/Share /u:XXXX /v:none.com /w:1920 /h:1080 +auto-reconnect /auto-reconnect-max-retries:5 /p:XXX +clipboard /log-level:trace

After resume remote session press Win 2x :)

**Comment by weaponrex** [comment]
I have same issue:
version: This is FreeRDP version 2.1.1 (n/a)
on: Ubuntu 18.04.4 LTS

1) I connect from ubuntu to windows 10 computer using: xfreerdp /multimon /u:xxx /p:xxx /v:xxx
2) Starts on Win10 fullscreen, then ctrl+alt+enter to see ubuntu again as expected.
3) Now I press ctrl+alt+enter to go to full screen again as expected.
4) Then pressing enter alone toggles fullscreen off! NOT as expected.

smart-sizing does help
pressing the windows key twice didn't help
the num-pad enter doesn't trigger this bug, so for now this is my work-around.

**Comment by kotipalvelu** [comment]
> I have same issue:
> version: This is FreeRDP version 2.1.1 (n/a)
> on: Ubuntu 18.04.4 LTS
> 
>     1. I connect from ubuntu to windows 10 computer using: xfreerdp /multimon /u:xxx /p:xxx /v:xxx
> 
>     2. Starts on Win10 fullscreen, then ctrl+alt+enter to see ubuntu again as expected.
> 
>     3. Now I press ctrl+alt+enter to go to full screen again as expected.
> 
>     4. Then pressing enter alone toggles fullscreen off! NOT as expected.
> 
> 
> [edit] smart-sizing doesn't help since I use multiple monitors
> pressing the windows key twice didn't help
> the num-pad enter doesn't trigger this bug, so for now this is my work-around.
> [edit] Or restarting the connection to avoid the bug.

I can confirm this behavior. My system is also Ubuntu 18.04 login to Gnome+Wayland -variant of desktop. I am logging to W10 client which is running on host as a Virtualbox session.

**Comment by lghtman** [comment]
I'm having exactly the same problem as @weaponrex and @kotipalvelu, but connecting to a physical Windows 10 machine instead of a VM.

**Comment by Jaculabilis** [comment]
I am having the same problem as @weaponrex and also cannot use smart-sizing due to using multiple monitors. Pressing Win twice immediately after using ctrl-alt-enter to fullscreen the remote window does **not** work for me. Restarting the connection does work, as the issue only arises after I use ctrl-alt-enter to un-fullscreen once, but this is not a very good solution because it messes with all the windows on the remote machine. Pressing the numpad enter also works, but the muscle memory to press the normal enter is too strong.

**Comment by akallabeth** [comment]
I´ve reworked some parts of the input handling with https://github.com/FreeRDP/FreeRDP/pull/6323
Any change with that?

**Comment by yashi** [comment]
It seems like this issue happens only with `-grab-keyboard` in a Wayland session with `Xwayland`.  And No, akallabeth:x11_kbd_mapping doesn't fix it.

To reproduce,

1. Run a GNOME wayland session
2. Run xfreerdp with `-grab-keyboard` and connect to a Windows
3. Switch to a second workspace
4. Run xlogo and make sure it gets focused
5. Go back to the previous workspace and hit enter in the xfreerdp

An odd thing is that with xeyes, xclock, xload, or oclock in the second workspace doesn't trigger this bug.  But xlogo, xcalc, or xconsole does.  The diff AFAICS is that the apps which trigger this bug get drop shadow and the title bar changes color when it gets focused.  The apps which don't trigger don't get any visual indication for "focused".

A native wayland app, like gnome-terminal or gedit, doesn't trigger this bug.  Google Chrome, or any other non-native / x11 app, triggers.

This is on Debian Sid with

gnome-shell    3.36.3-1
mutter         3.36.3-1
x11-apps       7.7+8
xwayland       2:1.20.8-2

I've tested against FreeRDP "2.1.2", "stable-2.0", "master", and "akallabeth:x11_kbd_mapping".  All behave the same.

Hope this helps.

**Comment by sunyz** [comment]
ubuntu 18.04 , xfreerdp2-x11 , I am having the same problem
Press Win 2x don't work too.
Finally, I tried a strange but useful method：
after use ctrl+alt+enter resume， Press [**Alt + F8**], and MUST be **Right Alt** on your keyboard, Left Alt cannot work.

**Comment by vtregub** [comment]
The same behavior
FreeRDP version 2.1.1
Ubuntu 18.04.5 LTS

Single press of RightAlt helps after returning to fullscreen

**Comment by alexhuot1** [comment]
I've got the same problem. 
All the above solutions don't work. Some of them worked but only once or twice.

Manjaro (Kernel 5.12.1)
FreeRDP version 2.3.2

**Comment by alexhuot1** [comment]
Turns out... I've switched to gnome (x-org) instead of Wayland... and I did not have the problem since. I'll keep you up to date!

**Comment by wanghongzhi0206** [comment]
Ubuntu 18.04.5 LTS
2.0.0-rc3 OK
from 2.0.0-rc4 NG

I compared the code of rc3 and rc4. Starting from rc4, function xf_keyboard_update_modifier_keys(xfc) was added to  xf_keyboard.c, and it was OK after deleting it.

**Comment by pavel-zhigulin** [comment]
Ubuntu 20.04.3 LTS
This is FreeRDP version 2.2.0 (n/a)

Still have the issue. I use following command to connect 

```bash
wlfreerdp /audio-mode:0 /bpp:24 /f  /microphone /p:password /gfx:RFX -glyph-cache +toggle-fullscreen /floatbar /sound -themes /u:user /v:<ip> -wallpaper
```
As workaround I use following:

1. Press Win + Esc, which activates host hotkeys
2. Press Alt + Tab to switch to any Ubuntu window
3. Terminate wlfreerdp from ubuntu terminal

**Comment by akallabeth** [comment]
@pavel-zhigulin you´re talking about `wlfreerdp`, this issue is for `xfreerdp` (different clients)

**Comment by recolic** [comment]
similar problem. I'm using xfreerdp3, using `ctrl+alt+up/down/left/right` to switch between workspaces.

When I press enter, it triggered fullscreen. When I press enter again, it exited fullscreen.

**Comment by recolic** [comment]
Solution: I'm using `xfreerdp -disable-grab-keyboard -disable-toggle-fullscreen` and pretty happy.
